### PR TITLE
Converts to Typescript

### PR DESCRIPTION
**Pull request by orta**
Phew - this looks like everything. 

Converts the app to use typescript, there's a tonne of TSLint warnings but no errors. All of the pages but react-native-web load, RNW shows some of it, but is a red screen.

screen shot 2017-01-03 at 1 01 33 pm

etc

screen shot 2017-01-03 at 1 01 42 pm

No intermediate files, it pipes directly from TypeScript to Babel to pure JS

### CODE REVIEW

File: app/containers/react-inline-css/artist/artwork.tsx [hunk 1]
```diff
@@ -1,7 +1,13 @@
...
-import Relay from 'react-relay'
+import * as React from "react"
+import * as Relay from "react-relay"
 
-export class Artwork extends React.Component {
+interface Props {
+  artwork: any
+}
+interface State {
+ }
```
Comment (alloy) [inline app/containers/react-inline-css/artist/artwork.tsx]: Holy white space, batman!

**Comment by orta** [comment]
WRT `"` vs `'` the default is `"` in Typescript, reasoning is here - https://github.com/Microsoft/TypeScript/issues/5811#issuecomment-160187924

tslint is generally base on https://github.com/Microsoft/TypeScript/wiki/Coding-guidelines

File: app/routes.tsx [hunk 2]
```diff
@@ -1,31 +1,37 @@
...
 app.use(artsyRelayMiddleware)
 
+app.get("/", (req: any, res: any, next: any) => {
+  res.send(`<html><body><ul>
+  <li><a href='/pure-react/artist/banksy'>pure-react</a></li>
+  <li><a href='/react-inline-css/artist/banksy'>react-inline-css</a></li>
+  <li><a href='/react-aphrodite/artist/banksy'>react-aphrodite</a></li>
+  <li><a href='/react-native-web/artist/banksy'>react-native-web</a></li>
+  <li><a href='/react-jss/artist/banksy'>react-jss</a></li>
+  `)
```
Comment (alloy) [inline app/routes.tsx]: Where. Are. The. Close. Tags???!!!

Giphy

File: webpack.js [hunk 3]
```diff
@@ -0,0 +1,6 @@
+
+module.exports = function(content, thing) {
+  console.log(content)
+  console.log(thing)
+  return content
+}
```
Comment (alloy) [inline webpack.js:6]: This one can go.

**Review by alloy — CHANGES_REQUESTED** [review]

File: app/components/artist/header.tsx [hunk 4]
```diff
@@ -1,34 +1,36 @@
...
+import * as React from "react"
+import * as Relay from "react-relay"
 
-type ArtistHeaderProps = { artist: ArtistItem }
+interface Props {
+  artist: any
+}
+interface State {
+      following: boolean | null,
+    followersCount: number,
```
Comment (alloy) [inline app/components/artist/header.tsx]: pipe-r indentation!

**Review by alloy — APPROVED** [review]

**Comment by alloy** [comment]
Types all the way! 🐢

**Comment by alloy** [comment]
@orta oops, this stray webpack loader still made it in https://github.com/alloy/relational-theory/blob/master/webpack.js

**Pull request merged**